### DISCUSSION TITLE
Reusing caches from Merge Queue checks?

### DISCUSSION THREAD

**Discussion by siketyan** [community/community#160583] [Actions]
I am using actions/cache in a Merge Queue enabled repository. The Merge Queue checks work instead of checks by push trigger.

Usually actions/cache reuses caches from main branch when running on a topic branch or pull request merge (`refs/pull/<number>/merge`). However, caches are not reused from Merge Queue checks (`gh-readonly-queue/main/<ref>`). It would be great if I can assume the `gh-readonly-queue/main/*` is a global scope; any other topic branches can reuse from caches from the refs.

Is there any workaround or something for this?

**Comment by tsvetomir** [comment]
Are you sure you have a populated cache in your main branch?

I was having a similar issue, but it turned out that I've never added a cache entry from the main branch. Once this was sorted, the cache was accessible to the action in the merge queue.

**Reply by jdelStrother** [reply]
I believe the problem @siketyan & I are are experiencing is different - the merge queue can re-use a cache from the main branch, but it then saves a cache on the `gh-readonly-queue/main/foo` branch, which is effectively useless. Any work done to generate that cache can't be re-used by the main branch or a new `gh-readonly-queue/main/bar` branch, so they have to do it all over again.

**Reply by siketyan** [reply]
@jdelStrother Indeed that's the problem I am experiencing. We need to run the workflow both on merge queue trigger and push trigger just for caching, or we can't get any cache on each run.

**Reply by siketyan** [reply]
AFAIK every commit in a merge queue must be linear, and safe to skip checks after they passed the CI and got merged. We _can_ also run it again on the main branch for cache, wasting computing resources and our money.

It will be great if we can share the cache _at least_ among runs on the queue. As we don't need to run push trigger on the main branch anymore if merge queue is enforced, I actually want to use the cache on pull requests that is populated on the merge queue for the base branch.

**Reply by jdelStrother** [reply]
Yeah. I'm wondering if I need to replace some of our actions/cache usage with actions/{upload,download}-artifact to get around this
